pages/words/index.js

```diff
@@ -3,13 +3,15 @@ import Heading from "@/components/PageHeading";
 import { MainContent } from "@/components/StyledComponents/MainContent";
 
 export default function WordList({ wordList, databaseIsLoading }) {
-  return (
-    <MainContent>
-      <Heading PageTitle="Your Saved Words" />
-      <EntriesContainer
-        wordList={wordList}
-        databaseIsLoading={databaseIsLoading}
-      />
-    </MainContent>
-  );
+  if (wordList) {
```

Comment:
Since the rendering depends on `wordList` only, you can make the check in the parent component - `{ wordlList && <WordList ...> }`